gatsby-node.js

```diff
@@ -1,51 +1,52 @@
 /* eslint "no-console": "off" */
 
-const path = require("path");
-const _ = require("lodash");
-const moment = require("moment");
-const siteConfig = require("./data/SiteConfig");
+const path = require('path')
+const _ = require('lodash')
+const moment = require('moment')
+const siteConfig = require('./data/SiteConfig')
+
+const hasOwnProperty = (obj, property) => Object.prototype.hasOwnProperty.call(obj, property)
 
 exports.onCreateNode = ({ node, actions, getNode }) => {
-  const { createNodeField } = actions;
-  let slug;
-  if (node.internal.type === "MarkdownRemark") {
-    const fileNode = getNode(node.parent);
-    const parsedFilePath = path.parse(fileNode.relativePath);
-    if (
-      Object.prototype.hasOwnProperty.call(node, "frontmatter") &&
-      Object.prototype.hasOwnProperty.call(node.frontmatter, "title")
-    ) {
-      slug = `/${_.kebabCase(node.frontmatter.title)}`;
-    } else if (parsedFilePath.name !== "index" && parsedFilePath.dir !== "") {
-      slug = `/${parsedFilePath.dir}/${parsedFilePath.name}/`;
-    } else if (parsedFilePath.dir === "") {
-      slug = `/${parsedFilePath.name}/`;
-    } else {
-      slug = `/${parsedFilePath.dir}/`;
-    }
+  const { createNodeField } = actions
+
+  // work out and add the slug and date fields to the MarkdownRemark nodes
+  if (node.internal.type === 'MarkdownRemark') {
+    const fileNode = getNode(node.parent)
+    const { dir, name } = path.parse(fileNode.relativePath)
+
+    const hasFrontMatter = hasOwnProperty(node, 'frontmatter')
+    const hasSlug = hasFrontMatter && hasOwnProperty(node, 'slug')
+    const hasTitle = hasFrontMatter && hasOwnProperty(node, 'title')
+
+    const slug =
+      (hasSlug && `/${_.kebabCase(node.frontmatter.slug)}`) ||
+      (hasTitle && `/${_.kebabCase(node.frontmatter.title)}`) ||
+      (name !== 'index' && dir !== '' && `/${dir}/${name}/`) ||
+      (dir === '' && `/${name}/`) ||
+      `/${dir}/`
```

Comment:
Pulling all the if else logic out into some boolean logic for the `slug` seems clearer to me but not fussed to move back into if statements.

Reply:
Could maybe be a helper function?

Reply:
Yeah a function would be better still!